### ISSUE TITLE
Deserialising messages returned from Search fail if they have attachments with timestamps

### ISSUE THREAD

**Issue by ProductiveRage** [stoiveyp/Slack.NetStandard#42]
The timestamp value from the API is the Slack timestamp format (ie. "**{epoch-seconds}.{identifier}**") but the "Timestamp" property on the `Attachment` class is a `long?`. So if there is a non-null timestamp value for an attachment then a serialisation exception is thrown.

Example timestamp returned: "1495992386.240336"

Error is that this value can't be deserialised to `Int64`.

The `Message` and `File` classes both use the `Timestamp` class for their "Timestamp" property, it seems like Attachment is the odd one out.

As a workaround, I've written this Json.NET type converter:

```
private sealed class SlackTimestampToNullableLongConverter : JsonConverter<long?>
{
    public static SlackTimestampToNullableLongConverter Instance { get; } = new();
    private SlackTimestampToNullableLongConverter() { }

    public override long? ReadJson(JsonReader reader, Type objectType, [AllowNull] long? existingValue, bool hasExistingValue, JsonSerializer serializer)
    {
        if ((reader.Value is long) || (reader.Value is long?))
            return (long?)reader.Value;

        if (reader.Value is null)
            return null;

        // A Slack timestamp is returned in their API response in the form "1495992386.240336" which maps nicely to a Timestamp instance, such as the
        // Message class has as this has an "EpochSeconds" (long) property and an "Identifier" (string) property and I believe that the two segments
        // are used in combination to describe not only a point in time but a unique-to-a-channel point in time (presumably for ordering purposes)
        // - see https://github.com/slackhq/slack-api-docs/issues/7 for more info.
        //
        // However, the Attachment class doesn't use the Timestamp class as its "Timestamp" property, it uses a nullable long trying to deserialise
        // from something like "1495992386.240336" to a nullable long will fail. Adding this type converter to the SlackWebApiClient can workaround
        // the issue until the data model is fixed up.
        if ((reader.Value is string s) && long.TryParse(s.Split('.')[0], out var parsedValue))
            return parsedValue;
                     
        throw new NotImplementedException();
    }

    public override void WriteJson(JsonWriter writer, [AllowNull] long? value, JsonSerializer serializer)
    {
        if (value.HasValue)
            writer.WriteValue(value.Value);
        else
            writer.WriteNull();
    }
}
```

.. and when I instantiate the `SlackWebApiClient`, I do it like this:

```
var client = new SlackWebApiClient(accessToken);
client.Serializer.Converters.Add(SlackTimestampToNullableLongConverter.Instance);
```

I _presume_ that this is an issue with the type of the property on the `Attachment` class but I hadn't tried using your library before today and so you would know much better than I!

**Comment by stoiveyp** [comment]
Hi @ProductiveRage - thanks for raising this.

The documentation specifically says attachment timestamps are supposed to be integers, which is why this one was different to the others.

I've updated the Attachment type to use the same Timestamp class as the other objects - this is available now in v2.10.1 and v2.11.0-beta2 which have just been published to NuGet.

Hope this helps.

**Comment by ProductiveRage** [comment]
Well, certainly can't argue with that sort of turnaround time! :) Updated to v2.10.1 and it has indeed fixed it - thanks again!
